src/backend/backend.hpp

```diff
@@ -41,13 +41,14 @@ namespace argo {
 	namespace backend {
 		/**
 		 * @brief initialize backend
-		 * @param size the size of the global memory to initialize
+		 * @param argo_size the size of the global memory to initialize
+		 * @param cache_size the size of the cache used by ArgoDSM in number of bytes, may be unused in some backends
 		 * @warning the signature of this function may change
 		 * @todo maybe this should be tied better to the concrete memory
 		 *       allocated rather than a generic "initialize this size"
 		 *       functionality.
 		 */
-		void init(std::size_t size);
+		void init(std::size_t argo_size, std::size_t cache_size);
```

Comment:
Should `cache_size` have a default value?

Reply:
Would that be for backwards compatibility, or why should the backend have a default value for this?

Reply:
I don't remember any more, but I would make more sense to have the default argument in the frontend part.

Reply:
well, the frontend code hard-codes the value to 1G..